pkg/kubelet/config/common.go

```diff
@@ -73,6 +73,21 @@ func applyDefaults(pod *api.Pod, source string, isFile bool, nodeName types.Node
 	// The generated UID is the hash of the file.
 	pod.Annotations[kubetypes.ConfigHashAnnotationKey] = string(pod.UID)
 
+	if isFile {
+		// Applying the default Taint tolerations to static pods,
+		// so they are not evicted when there are node problems.
```

Comment:
I thought about this, but still choose to make a cautious decision to hardcoded all NoExecute taints. But I think you guys are right on this.